From: Linus Lüssing <linus.luessing@c0d3.blue>
Date: Mon, 27 Nov 2023 08:19:30 +0100
Subject: Revert "batman-adv: Use sizeof_field() macro"

The sizeof_field() macro was introduced in Linux 4.16 and initially
used/introduced in batman-adv v2020.0.

To be able to build batman-adv v2023.3 on OpenWrt 19.07 /
backports-4.19.237-1 we need to partially revert the according
commit in batman-adv, depending on the backports version used, as
backports-4.19 does not provide this function either.

This reverts commit 44fa6734a37674498a0e61a2e60f3b8d413d9f66.

Signed-off-by: Linus Lüssing <linus.luessing@c0d3.blue>

diff --git a/net/batman-adv/main.c b/net/batman-adv/main.c
index 8a536e1d9e8547abcb4649872d4cbb78168ed12c..6ddf5c7b906b2477365b7e1a9cfbe431b0ea1dd2 100644
--- a/net/batman-adv/main.c
+++ b/net/batman-adv/main.c
@@ -528,7 +528,11 @@ static void batadv_recv_handler_init(void)
 	BUILD_BUG_ON(sizeof(struct batadv_tvlv_tt_change) != 12);
 	BUILD_BUG_ON(sizeof(struct batadv_tvlv_roam_adv) != 8);
 
+#ifdef HAVE_BACKPORTS_LESS_5_10 // UGLY_HACK_START
+	i = FIELD_SIZEOF(struct sk_buff, cb);
+#else // UGLY_HACK_NEW
 	i = sizeof_field(struct sk_buff, cb);
+#endif // UGLY_HACK_STOP
 	BUILD_BUG_ON(sizeof(struct batadv_skb_cb) > i);
 
 	/* broadcast packet */
